### PR TITLE
MapShed Manual Entry: Add Animals Tab

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Adds Animals tab to Manual Entry modal. Also adds the ability to have dropdown fields. Their values are grayed until the user edits them, when the undo button is visible, which behaves exactly as for the text boxes.

Also fixes the modal itself so it doesn't spawn multiple copies.

Connects #2960 

### Demo

image

### Notes

The dropdowns look different than the text boxes in terms of fill, border, and shadow, but that is browser dependent. I kept them the same width, and made sure the undo button didn't overlap any of the native UI in different browsers.

## Testing Instructions

* Check out this branch and `bundle`
* Go to a MapShed project's scenario and open the settings modal
* Switch to the Animals tab. Ensure the fields match the original issue.
* Try editing values. Ensure they behave as expected in the GMS output.
* Ensure they load correctly for saved scenarios
* Ensure you can close and open the entry modal multiple times and switch tabs correctly

### CODE REVIEW

File: src/mmw/js/src/modeling/gwlfe/entry/models.js [hunk 1]
```diff
@@ -3,10 +3,16 @@
 var Backbone = require('../../../../shim/backbone'),
     GwlfeModificationModel = require('../../models').GwlfeModificationModel;
 
+var ENTRY_FIELD_TYPES = {
+    NUMERIC: 'NUMERIC',
+    YESNO: 'YESNO',
```
Comment (kellyi) [inline src/mmw/js/src/modeling/gwlfe/entry/models.js]: This is probably fine but it seems like BOOLEAN would be a decent alternative name.

Comment (kellyi) [reply src/mmw/js/src/modeling/gwlfe/entry/models.js]: Consider `Object.freeze`ing this enum.

**Comment by kellyi** [comment]
Tried out this AOI with two adjusted scenarios:

screen shot 2018-09-13 at 5 53 16 pm

One scenario has 150000 swine allowed to graze:

screen shot 2018-09-13 at 5 54 01 pm

The other has 0 swine:

screen shot 2018-09-13 at 5 54 28 pm

GMS file difference is:

```diff
diff --git a/15000grazingswine.gms b/nograzingswine.gms
index b734e6c..4d46984 100644
--- a/15000grazingswine.gms
+++ b/nograzingswine.gms
@@ -149,7 +149,7 @@ Dairy Cows,159,Y,640,0.44,0.07,100000000000
 Beef Cows,81,Y,360,0.31,0.09,100000000000
 Broilers,178,N,0.9,1.07,0.3,140000000
 Layers,434,N,1.8,0.85,0.29,140000000
-Hogs/Swine,15000,Y,61,0.48,0.15,11000000000
+Hogs/Swine,0,N,61,0.48,0.15,11000000000
 Sheep,439,Y,50,0.37,0.1,12000000000
 Horses,695,Y,500,0.28,0.06,420000000
 Turkeys,95,N,6.8,0.59,0.2,95000000
```

**Comment by rajadain** [comment]
Since it's called `YesOrNo` in GWLF-E:

https://github.com/WikiWatershed/gwlf-e/blob/465a6ae433de25bd77aebb19401dca6c51f9b168/gwlfe/enums.py#L7-L9

and given the 🚀 🐦  nature of these times, I'm going to leave it as it and merge it. Thanks for taking a look!

**Pull request merged**